connexion/apis/starlette_api.py

```diff
@@ -0,0 +1,333 @@
+            query=query,
+            headers=headers,
+            body=body,
+            json_getter=lambda: cls.jsonifier.loads(body),
+            files={},
+        )
+
+    @classmethod
+    async def get_response(cls, response, mimetype=None, request=None):
+        """Get response.
+        This method is used in the lifecycle decorators
+
+        :type response: starlette.responses.Response | (Any,) | (Any, int) | (Any, dict) | (Any, int, dict)
+        :rtype: starlette.responses.Response
+        """
+        while asyncio.iscoroutine(response):
+            response = await response
+        url = str(request.url) if request else ""
+
+        return cls._get_response(
+            response, mimetype=mimetype, extra_context={"url": url}
+        )
+
+    @classmethod
+    def _is_framework_response(cls, response):
+        """ Return True if `response` is a framework response class """
+        return isinstance(response, Response)
+
+    @classmethod
+    def _framework_to_connexion_response(cls, response, mimetype):
+        """ Cast framework response class to ConnexionResponse used for schema validation """
+
+        # FileResponse and StreamingResponse do not a `body` (yet)
+        body = None
+        if hasattr(response, 'body'):
+            body = response.body
+
+        return ConnexionResponse(
+            status_code=response.status_code,
+            mimetype=mimetype,
+            content_type=response.media_type,
+            headers=response.headers,
+            body=body,
+        )
+
+    @classmethod
+    def _connexion_to_framework_response(cls, response, mimetype, extra_context=None):
+        """ Cast ConnexionResponse to framework response class """
+        return cls._build_response(
+            mimetype=response.mimetype or mimetype,
+            status_code=response.status_code,
+            content_type=response.content_type,
+            headers=response.headers,
+            data=response.body,
+            extra_context=extra_context,
+        )
+
+    @classmethod
+    def _build_response(
+        cls,
+        data,
+        mimetype,
+        content_type=None,
+        headers=None,
+        status_code=None,
+        extra_context=None,
+    ):
+        if cls._is_framework_response(data):
+            raise TypeError(
+                "Cannot return starlette.responses.Response in tuple. Only raw data can be returned in tuple."
+            )
+        data, status_code, serialized_mimetype = cls._prepare_body_and_status_code(
+            data=data,
+            mimetype=mimetype,
+            status_code=status_code,
+            extra_context=extra_context,
+        )
+
+        content_type = content_type or mimetype or serialized_mimetype
+
+        if content_type is None:
+            if isinstance(data, str):
+                content_type = "text/plain"
+            elif isinstance(data, bytes):
+                content_type = "application/octet-stream"
```

Comment:
Would be nice to add another option to check if the data is a `dict` or `list` then use [`JSONResponse`](https://www.starlette.io/responses/#jsonresponse). What do you think?

Reply:
Merged relevant changes, tests pass, should be fine :)